skfp/model_selection/splitters/pubchem_split.py

```diff
@@ -0,0 +1,456 @@
+    is True, lists of indices are returned instead of actual data.
+
+    References
+    ----------
+    .. [1] `Kim S, Thiessen PA, Cheng T, Yu B, Bolton EE.
+        "An update on PUG-REST: RESTful interface for programmatic access to PubChem."
+        Nucleic Acids Res. 2018 Jul 2;46(W1):W563-W570.
+        <https://doi.org/10.1093/nar/gky294>`_
+    """
+    years = _get_pubchem_years(data, n_jobs)
+
+    data_df = pd.DataFrame({"idx": list(range(len(data))), "year": years})
+
+    if not_found_behavior != "remove":
+        not_found_mols = list(data_df[data_df["year"].isna()].idx)
+
+    data_df = data_df[~data_df["year"].isna()]
+    data_df = data_df.sort_values("year")
+
+    train_size, valid_size, test_size = validate_train_valid_test_split_sizes(
+        train_size, valid_size, test_size, len(data)
+    )
+
+    train_idxs: list[int] = list(data_df.iloc[:train_size].idx)
+    valid_idxs: list[int] = list(data_df.iloc[train_size : train_size + valid_size].idx)
+    test_idxs: list[int] = list(data_df.iloc[train_size + valid_size :].idx)
+
+    if not_found_behavior == "train":
+        train_idxs.extend(not_found_mols)
+
+    if not_found_behavior == "test":
+        test_idxs.extend(not_found_mols)
+
+    if return_indices:
+        train_subset = train_idxs
+        valid_subset = valid_idxs
+        test_subset = test_idxs
+    else:
+        train_subset = get_data_from_indices(data, train_idxs)
+        valid_subset = get_data_from_indices(data, valid_idxs)
+        test_subset = get_data_from_indices(data, test_idxs)
+
+    ensure_nonempty_subset(train_subset, "train")
+    ensure_nonempty_subset(valid_subset, "validation")
+    ensure_nonempty_subset(test_subset, "test")
+
+    if additional_data:
+        additional_data_split: list[Sequence[Any]] = split_additional_data(
+            list(additional_data), train_idxs, valid_idxs, test_idxs
+        )
+        return train_subset, valid_subset, test_subset, *additional_data_split
+    else:
+        return train_subset, valid_subset, test_subset
+
+
+def _get_pubchem_years(
+    data: Sequence[Union[str, Mol]], n_jobs: int = 5
+) -> list[list[int]]:
+    """
+    Get first literature publication year from PubChem for a list of molecules, either
+    as SMILES strings or RDKit Mol objects.
+
+    Parameters
+    ----------
+    data : sequence
+        A sequence representing either SMILES strings or RDKit `Mol` objects.
+    """
+    data = list(
+        map(
+            lambda mol: (
+                MolToSmiles(mol, canonical=True, isomericSmiles=True)
+                if mol is Mol
+                else mol
+            ),
+            data,
+        )
+    )
+
+    print("Converting SMILES to CIDs")
+    cids = run_in_parallel(get_cid_for_smiles, data, n_jobs)
+
+    print("Searching for CIDs literature")
+    years = run_in_parallel(get_earliest_publication_date, cids, n_jobs)
+
+    return years
+
+
+def get_cid_for_smiles(smiles: str) -> Optional[str]:
```

Comment:
I think this can have `_` at the beginning, it's pretty internal function. And then docstring can be a minimal comment, we don't need whole `Parameters` and `Return` format for internal functions.